config/config_test.go

```diff
@@ -654,3 +656,70 @@ func TestTimeToLiveDuration(t *testing.T) {
 	require.NotNil(t, err)
 	require.EqualError(t, err, "time: unknown unit \"d\" in duration \"2d\"")
 }
+
+func TestNewConfig(t *testing.T) {
+	tmpFile, err := os.CreateTemp("/tmp", "tmpfile")
+	require.Nil(t, err)
+	defer os.RemoveAll(tmpFile.Name())
+
+	f, err := os.OpenFile(tmpFile.Name(), os.O_WRONLY, 0600)
+	require.Nil(t, err)
+	defer f.Close()
+
+	defaultCfg := getDefaultConfig(t)
+	enc := toml.NewEncoder(f)
+	err = enc.Encode(defaultCfg)
+	require.Nil(t, err)
+
+	cfg, err := NewConfig(tmpFile.Name())
+	require.Nil(t, err)
+	require.NotNil(t, cfg)
+	require.Equal(t, &defaultCfg, cfg)
+}
+
+func TestNewConfigNoConfigDir(t *testing.T) {
+	tmpFile, err := os.CreateTemp("/tmp", "tmpfile")
+	require.Nil(t, err)
+	defer os.RemoveAll(tmpFile.Name())
+
+	f, err := os.OpenFile(tmpFile.Name(), os.O_WRONLY, 0600)
+	require.Nil(t, err)
+	defer f.Close()
+
+	defaultCfg := getDefaultConfig(t)
+	defaultCfg.Default.ConfigDir = ""
+	enc := toml.NewEncoder(f)
+	err = enc.Encode(defaultCfg)
+	require.Nil(t, err)
+
+	cfg, err := NewConfig(tmpFile.Name())
+
+	if _, statErr := os.Stat(DefaultConfigDir); statErr != nil {
+		require.NotNil(t, err)
+		require.Regexp(t, fmt.Sprintf("stat %s: no such file or directory", DefaultConfigDir), err.Error())
+		require.Nil(t, cfg)
+	} else {
+		require.NotNil(t, cfg)
+		defaultCfg.Default.ConfigDir = DefaultConfigDir
+		require.Equal(t, &defaultCfg, cfg)
+		require.Nil(t, err)
+	}
+}
+
+func TestNewConfigInvalidToml(t *testing.T) {
+	cfg, err := NewConfig("this is not a file path")
+	require.Nil(t, cfg)
+	require.NotNil(t, err)
+	require.Regexp(t, "decoding toml", err.Error())
+}
+
+func TestNewConfigInvalidConfig(t *testing.T) {
```

Comment:
An empty file is actually not an invalid ```toml``` config. The struct will still be populated with default values, but it will not pass ```Validate()``` because we expect that the user will set some values manually. For some options we have defaults, but some we expect to get from the user.

It would be useful to write some bogus data into that file and call ```NewConfig()``` on that. This way we test how the function will behave when the config is corrupt in some way.